gazelle/resolve.go

```diff
@@ -55,8 +97,19 @@ func (l *swiftLang) Resolve(
 	// Try to resolve to targets in this project.
 	externalModules := make([]string, 0, len(swiftImports))
 	for _, imp := range swiftImports {
+		importSpec := resolve.ImportSpec{Lang: swiftLangName, Imp: imp}
+		if l, ok := resolve.FindRuleWithOverride(c, importSpec, "swift"); ok {
+			addToDeps(l)
+			rr.AddLocal(imp, []resolve.FindResult{
+				{
+					Label:  l,
+					Embeds: nil, // TODO: This might be broken -- not sure what to put here.
```

Comment:
Is this really a TODO? Can you explain the context?

Reply:
@cgrindel this was left from when I was initially trying to implement the feature and I wasn't sure what purpose the embeds served and if they were necessary. After some investigation, I found that the "embeds" are kind of a Go-specific concept in Gazelle. If target A embeds target B, only A will be indexed and can be resolved as a dependency by other targets.

There is a method called Embeds on the Resolver interface in gazelle which supplies these, but the implementation for the proto language just returns nil from this method.

```
	// Embeds returns a list of labels of rules that the given rule embeds. If
	// a rule is embedded by another importable rule of the same language, only
	// the embedding rule will be indexed. The embedding rule will inherit
	// the imports of the embedded rule.
	Embeds(r *rule.Rule, from label.Label) []label.Label
```

I don't actually see an implementation of this method at all in rules_swift_package_manager so I'm unsure how it's able to implement the Resolver interface (which is embedded in the Language interface)?

https://github.com/bazelbuild/bazel-gazelle/blob/f5a5c5dc90b4d6a39f0f47d770eba27c6c1c1877/resolve/index.go#L47

In any case, it seems fine to pass nil here, as that's what the proto language does. I've removed the comment here and pushed the commit.

Reply:
We [start from language.BaseLang](https://github.com/cgrindel/rules_swift_package_manager/blob/4080822c6dc6aed07be6ba8459d9bcf222dcb51e/gazelle/lang.go#L13-L15). This provides reasonable defaults for Gazelle features that are not required for a plugin.

Reply:
Ah I missed that -- yeah then it's getting the default Embeds implementation which just returns nil:

```
func (b *BaseLang) Embeds(r *rule.Rule, from label.Label) []label.Label {
	return nil
}
```